core/primitives/src/bandwidth_scheduler.rs

```diff
@@ -82,6 +82,55 @@ pub struct BandwidthRequest {
 /// There are this many predefined values of bandwidth that can be requested in a BandwidthRequest.
 pub const BANDWIDTH_REQUEST_VALUES_NUM: usize = 40;
 
+/// Values of bandwidth that can be requested in a bandwidth request.
+/// When the nth bit is set in a request bitmap, it means that a shard is requesting the nth value from this list.
+/// The list is sorted, from smallest to largest values.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct BandwidthRequestValues {
+    pub values: [Bandwidth; BANDWIDTH_REQUEST_VALUES_NUM],
+}
+
+impl BandwidthRequestValues {
+    pub fn new(params: &BandwidthSchedulerParams) -> BandwidthRequestValues {
+        // values[-1] = base_bandwidth
+        // values[values.len() - 1] = max_shard_bandwidth
+        // values[i] = linear interpolation between values[-1] and values[values.len() - 1]
```

Comment:
For whatever reason I assumed it would be exponential. I guess it depends on the actual numbers. I'm totally fine with the simplest one to begin with, we can always tweak it later.

Reply:
Yeah exponential would probably be a bit better, but for now I went with the simplest option. The biggest problem with exponential is that it requires floats, and I'm scared of doing float operations in code that requires determinism. Maybe it's possible to simulate deterministic floats using ints or something?

Reply:
Adda a TODO about it